src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMapping.cs

```diff
@@ -20,34 +22,39 @@ public class PackageSourceMapping
         /// </summary>
         internal IReadOnlyDictionary<string, IReadOnlyList<string>> Patterns { get; }
 
-        private Lazy<SearchTree> SearchTree { get; }
+        private Lazy<SearchTree>? SearchTree { get; }
```

Comment:
Having a lazy that is also nullable feels weird. Although looking below, I see it's because this value will be null when `IsEnabled` is false. Maybe worth a comment?

Reply:
I agree it feels weird, but after I received feedback (https://github.com/NuGet/NuGet.Client/pull/5339#discussion_r1282180240), I saw the perspective of just letting it be `null` when we *know* there's no `SearchTree` due to `IsEnabled` being `false`. 

I like that this class is immutable, meaning that I can make assertions like this Lazy can declared in the constructor, and then there's no concern over stale state.